core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -873,4 +904,20 @@ private static Cache<String, AuthSession> newSessionCache(Map<String, String> pr
             (RemovalListener<String, AuthSession>) (id, auth, cause) -> auth.stopRefreshing())
         .build();
   }
+
+  private Cache<TableOperations, FileIO> newFileIOCloser() {
+    return Caffeine.newBuilder()
+        .scheduler(
```

Comment:
From the docs for `removalListener`:

> For a more prompt notification on expiration a scheduler(Scheduler) may be configured.

Do we need a scheduler? Isn't it fine if this happens when a new entry is added?

Reply:
I would probably go without a scheduler (I left this here from some previous experimentation). See my other comment in https://github.com/apache/iceberg/pull/7487#discussion_r1183971466 on that

Reply:
I've removed the scheduler